// @swimlane/ngx-datatable/release/components/footer/footer.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@swimlane/ngx-datatable/release/components/footer/footer.component';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/constants';
import * as import8 from '@angular/core/src/linker/component_factory';
import * as import9 from '@swimlane/ngx-datatable/release/components/footer/pager.component';
import * as import10 from './pager.component.ngfactory';
import * as import11 from '@angular/core/src/security';
export var Wrapper_DataTableFooterComponent = (function () {
    function Wrapper_DataTableFooterComponent() {
        this._changed = false;
        this.context = new import0.DataTableFooterComponent();
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
        this._expr_3 = import1.UNINITIALIZED;
        this._expr_4 = import1.UNINITIALIZED;
        this._expr_5 = import1.UNINITIALIZED;
        this._expr_6 = import1.UNINITIALIZED;
        this._expr_7 = import1.UNINITIALIZED;
        this._expr_8 = import1.UNINITIALIZED;
    }
    Wrapper_DataTableFooterComponent.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_DataTableFooterComponent.prototype.ngOnDestroy = function () {
        (this.subscription0 && this.subscription0.unsubscribe());
    };
    Wrapper_DataTableFooterComponent.prototype.check_footerHeight = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.footerHeight = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_DataTableFooterComponent.prototype.check_rowCount = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.rowCount = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_DataTableFooterComponent.prototype.check_pageSize = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.pageSize = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_DataTableFooterComponent.prototype.check_offset = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_3, currValue))) {
            this._changed = true;
            this.context.offset = currValue;
            this._expr_3 = currValue;
        }
    };
    Wrapper_DataTableFooterComponent.prototype.check_pagerLeftArrowIcon = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_4, currValue))) {
            this._changed = true;
            this.context.pagerLeftArrowIcon = currValue;
            this._expr_4 = currValue;
        }
    };
    Wrapper_DataTableFooterComponent.prototype.check_pagerRightArrowIcon = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_5, currValue))) {
            this._changed = true;
            this.context.pagerRightArrowIcon = currValue;
            this._expr_5 = currValue;
        }
    };
    Wrapper_DataTableFooterComponent.prototype.check_pagerPreviousIcon = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_6, currValue))) {
            this._changed = true;
            this.context.pagerPreviousIcon = currValue;
            this._expr_6 = currValue;
        }
    };
    Wrapper_DataTableFooterComponent.prototype.check_pagerNextIcon = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_7, currValue))) {
            this._changed = true;
            this.context.pagerNextIcon = currValue;
            this._expr_7 = currValue;
        }
    };
    Wrapper_DataTableFooterComponent.prototype.check_totalMessage = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_8, currValue))) {
            this._changed = true;
            this.context.totalMessage = currValue;
            this._expr_8 = currValue;
        }
    };
    Wrapper_DataTableFooterComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_DataTableFooterComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_DataTableFooterComponent.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_DataTableFooterComponent.prototype.subscribe = function (view, _eventHandler, emit0) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.page.subscribe(_eventHandler.bind(view, 'page')));
        }
    };
    return Wrapper_DataTableFooterComponent;
}());
var renderType_DataTableFooterComponent_Host = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, [], {});
var View_DataTableFooterComponent_Host0 = (function (_super) {
    __extends(View_DataTableFooterComponent_Host0, _super);
    function View_DataTableFooterComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DataTableFooterComponent_Host0, renderType_DataTableFooterComponent_Host, import6.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways);
    }
    View_DataTableFooterComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'datatable-footer', new import3.InlineArray2(2, 'class', 'datatable-footer'), rootSelector, null);
        this.compView_0 = new View_DataTableFooterComponent0(this.viewUtils, this, 0, this._el_0);
        this._DataTableFooterComponent_0_3 = new Wrapper_DataTableFooterComponent();
        this.compView_0.create(this._DataTableFooterComponent_0_3.context);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
        return new import8.ComponentRef_(0, this, this._el_0, this._DataTableFooterComponent_0_3.context);
    };
    View_DataTableFooterComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.DataTableFooterComponent) && (0 === requestNodeIndex))) {
            return this._DataTableFooterComponent_0_3.context;
        }
        return notFoundResult;
    };
    View_DataTableFooterComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (this._DataTableFooterComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange)) {
            this.compView_0.markAsCheckOnce();
        }
        this.compView_0.internalDetectChanges(throwOnChange);
    };
    View_DataTableFooterComponent_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
        this._DataTableFooterComponent_0_3.ngOnDestroy();
    };
    View_DataTableFooterComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_DataTableFooterComponent_Host0;
}(import2.AppView));
export var DataTableFooterComponentNgFactory = new import8.ComponentFactory('datatable-footer', View_DataTableFooterComponent_Host0, import0.DataTableFooterComponent);
var styles_DataTableFooterComponent = [];
var renderType_DataTableFooterComponent = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, styles_DataTableFooterComponent, {});
export var View_DataTableFooterComponent0 = (function (_super) {
    __extends(View_DataTableFooterComponent0, _super);
    function View_DataTableFooterComponent0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DataTableFooterComponent0, renderType_DataTableFooterComponent, import6.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckOnce);
        this._expr_12 = import1.UNINITIALIZED;
        this._expr_13 = import1.UNINITIALIZED;
        this._expr_14 = import1.UNINITIALIZED;
    }
    View_DataTableFooterComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this._el_1 = import3.createRenderElement(this.renderer, parentRenderNode, 'div', import3.EMPTY_INLINE_ARRAY, null);
        this._text_2 = this.renderer.createText(this._el_1, '\n      ', null);
        this._el_3 = import3.createRenderElement(this.renderer, this._el_1, 'div', new import3.InlineArray2(2, 'class', 'page-count'), null);
        this._text_4 = this.renderer.createText(this._el_3, '', null);
        this._text_5 = this.renderer.createText(this._el_1, '\n      ', null);
        this._el_6 = import3.createRenderElement(this.renderer, this._el_1, 'datatable-pager', new import3.InlineArray2(2, 'class', 'datatable-pager'), null);
        this.compView_6 = new import10.View_DataTablePagerComponent0(this.viewUtils, this, 6, this._el_6);
        this._DataTablePagerComponent_6_3 = new import10.Wrapper_DataTablePagerComponent();
        this._text_7 = this.renderer.createText(null, '\n       ', null);
        this.compView_6.create(this._DataTablePagerComponent_6_3.context);
        this._text_8 = this.renderer.createText(this._el_1, '\n     ', null);
        this._text_9 = this.renderer.createText(parentRenderNode, '\n  ', null);
        var disposable_0 = import3.subscribeToRenderElement(this, this._el_6, new import3.InlineArray2(2, 'change', null), this.eventHandler(this.handleEvent_6));
        this._DataTablePagerComponent_6_3.subscribe(this, this.eventHandler(this.handleEvent_6), true);
        this.init(null, (this.renderer.directRenderer ? null : [
            this._text_0,
            this._el_1,
            this._text_2,
            this._el_3,
            this._text_4,
            this._text_5,
            this._el_6,
            this._text_7,
            this._text_8,
            this._text_9
        ]), [disposable_0]);
        return null;
    };
    View_DataTableFooterComponent0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import9.DataTablePagerComponent) && ((6 <= requestNodeIndex) && (requestNodeIndex <= 7)))) {
            return this._DataTablePagerComponent_6_3.context;
        }
        return notFoundResult;
    };
    View_DataTableFooterComponent0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_6_0_0 = this.context.pagerLeftArrowIcon;
        this._DataTablePagerComponent_6_3.check_pagerLeftArrowIcon(currVal_6_0_0, throwOnChange, false);
        var currVal_6_0_1 = this.context.pagerRightArrowIcon;
        this._DataTablePagerComponent_6_3.check_pagerRightArrowIcon(currVal_6_0_1, throwOnChange, false);
        var currVal_6_0_2 = this.context.pagerPreviousIcon;
        this._DataTablePagerComponent_6_3.check_pagerPreviousIcon(currVal_6_0_2, throwOnChange, false);
        var currVal_6_0_3 = this.context.pagerNextIcon;
        this._DataTablePagerComponent_6_3.check_pagerNextIcon(currVal_6_0_3, throwOnChange, false);
        var currVal_6_0_4 = this.context.pageSize;
        this._DataTablePagerComponent_6_3.check_size(currVal_6_0_4, throwOnChange, false);
        var currVal_6_0_5 = this.context.rowCount;
        this._DataTablePagerComponent_6_3.check_count(currVal_6_0_5, throwOnChange, false);
        var currVal_6_0_6 = this.context.curPage;
        this._DataTablePagerComponent_6_3.check_page(currVal_6_0_6, throwOnChange, false);
        if (this._DataTablePagerComponent_6_3.ngDoCheck(this, this._el_6, throwOnChange)) {
            this.compView_6.markAsCheckOnce();
        }
        var currVal_12 = this.context.footerHeight;
        if (import3.checkBinding(throwOnChange, this._expr_12, currVal_12)) {
            this.renderer.setElementStyle(this._el_1, 'height', ((this.viewUtils.sanitizer.sanitize(import11.SecurityContext.STYLE, currVal_12) == null) ? null : (this.viewUtils.sanitizer.sanitize(import11.SecurityContext.STYLE, currVal_12).toString() + 'px')));
            this._expr_12 = currVal_12;
        }
        var currVal_13 = import3.inlineInterpolate(2, '', this.context.rowCount.toLocaleString(), ' ', this.context.totalMessage, '');
        if (import3.checkBinding(throwOnChange, this._expr_13, currVal_13)) {
            this.renderer.setText(this._text_4, currVal_13);
            this._expr_13 = currVal_13;
        }
        var currVal_14 = !this.context.isVisible;
        if (import3.checkBinding(throwOnChange, this._expr_14, currVal_14)) {
            this.renderer.setElementProperty(this._el_6, 'hidden', currVal_14);
            this._expr_14 = currVal_14;
        }
        this.compView_6.internalDetectChanges(throwOnChange);
    };
    View_DataTableFooterComponent0.prototype.destroyInternal = function () {
        this.compView_6.destroy();
        this._DataTablePagerComponent_6_3.ngOnDestroy();
    };
    View_DataTableFooterComponent0.prototype.handleEvent_6 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'change')) {
            var pd_sub_0 = (this.context.page.emit($event) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_DataTableFooterComponent0;
}(import2.AppView));
