// @swimlane/ngx-datatable/release/components/footer/pager.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@swimlane/ngx-datatable/release/components/footer/pager.component';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/constants';
import * as import8 from '@angular/core/src/linker/component_factory';
import * as import9 from '@angular/core/src/linker/view_container';
import * as import10 from '../../../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import11 from '@angular/core/src/linker/template_ref';
import * as import12 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import13 from '@angular/common/src/directives/ng_for';
export var Wrapper_DataTablePagerComponent = (function () {
    function Wrapper_DataTablePagerComponent() {
        this._changed = false;
        this.context = new import0.DataTablePagerComponent();
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
        this._expr_3 = import1.UNINITIALIZED;
        this._expr_4 = import1.UNINITIALIZED;
        this._expr_5 = import1.UNINITIALIZED;
        this._expr_6 = import1.UNINITIALIZED;
    }
    Wrapper_DataTablePagerComponent.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_DataTablePagerComponent.prototype.ngOnDestroy = function () {
        (this.subscription0 && this.subscription0.unsubscribe());
    };
    Wrapper_DataTablePagerComponent.prototype.check_pagerLeftArrowIcon = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.pagerLeftArrowIcon = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_DataTablePagerComponent.prototype.check_pagerRightArrowIcon = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.pagerRightArrowIcon = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_DataTablePagerComponent.prototype.check_pagerPreviousIcon = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.pagerPreviousIcon = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_DataTablePagerComponent.prototype.check_pagerNextIcon = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_3, currValue))) {
            this._changed = true;
            this.context.pagerNextIcon = currValue;
            this._expr_3 = currValue;
        }
    };
    Wrapper_DataTablePagerComponent.prototype.check_size = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_4, currValue))) {
            this._changed = true;
            this.context.size = currValue;
            this._expr_4 = currValue;
        }
    };
    Wrapper_DataTablePagerComponent.prototype.check_count = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_5, currValue))) {
            this._changed = true;
            this.context.count = currValue;
            this._expr_5 = currValue;
        }
    };
    Wrapper_DataTablePagerComponent.prototype.check_page = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_6, currValue))) {
            this._changed = true;
            this.context.page = currValue;
            this._expr_6 = currValue;
        }
    };
    Wrapper_DataTablePagerComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_DataTablePagerComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_DataTablePagerComponent.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_DataTablePagerComponent.prototype.subscribe = function (view, _eventHandler, emit0) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.change.subscribe(_eventHandler.bind(view, 'change')));
        }
    };
    return Wrapper_DataTablePagerComponent;
}());
var renderType_DataTablePagerComponent_Host = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, [], {});
var View_DataTablePagerComponent_Host0 = (function (_super) {
    __extends(View_DataTablePagerComponent_Host0, _super);
    function View_DataTablePagerComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DataTablePagerComponent_Host0, renderType_DataTablePagerComponent_Host, import6.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways);
    }
    View_DataTablePagerComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'datatable-pager', new import3.InlineArray2(2, 'class', 'datatable-pager'), rootSelector, null);
        this.compView_0 = new View_DataTablePagerComponent0(this.viewUtils, this, 0, this._el_0);
        this._DataTablePagerComponent_0_3 = new Wrapper_DataTablePagerComponent();
        this.compView_0.create(this._DataTablePagerComponent_0_3.context);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
        return new import8.ComponentRef_(0, this, this._el_0, this._DataTablePagerComponent_0_3.context);
    };
    View_DataTablePagerComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.DataTablePagerComponent) && (0 === requestNodeIndex))) {
            return this._DataTablePagerComponent_0_3.context;
        }
        return notFoundResult;
    };
    View_DataTablePagerComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (this._DataTablePagerComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange)) {
            this.compView_0.markAsCheckOnce();
        }
        this.compView_0.internalDetectChanges(throwOnChange);
    };
    View_DataTablePagerComponent_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
        this._DataTablePagerComponent_0_3.ngOnDestroy();
    };
    View_DataTablePagerComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_DataTablePagerComponent_Host0;
}(import2.AppView));
export var DataTablePagerComponentNgFactory = new import8.ComponentFactory('datatable-pager', View_DataTablePagerComponent_Host0, import0.DataTablePagerComponent);
var styles_DataTablePagerComponent = [];
var View_DataTablePagerComponent1 = (function (_super) {
    __extends(View_DataTablePagerComponent1, _super);
    function View_DataTablePagerComponent1(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DataTablePagerComponent1, renderType_DataTablePagerComponent, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_5 = import1.UNINITIALIZED;
        this._expr_6 = import1.UNINITIALIZED;
    }
    View_DataTablePagerComponent1.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.createRenderElement(this.renderer, null, 'li', new import3.InlineArray2(2, 'class', 'pages'), null);
        this._text_1 = this.renderer.createText(this._el_0, '\n        ', null);
        this._el_2 = import3.createRenderElement(this.renderer, this._el_0, 'a', new import3.InlineArray2(2, 'href', 'javascript:void(0)'), null);
        this._text_3 = this.renderer.createText(this._el_2, '', null);
        this._text_4 = this.renderer.createText(this._el_0, '\n      ', null);
        var disposable_0 = import3.subscribeToRenderElement(this, this._el_2, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_2));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._el_2,
            this._text_3,
            this._text_4
        ]), [disposable_0]);
        return null;
    };
    View_DataTablePagerComponent1.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_5 = (this.context.$implicit.number === this.parentView.context.page);
        if (import3.checkBinding(throwOnChange, this._expr_5, currVal_5)) {
            this.renderer.setElementClass(this._el_0, 'active', currVal_5);
            this._expr_5 = currVal_5;
        }
        var currVal_6 = import3.inlineInterpolate(1, '\n          ', this.context.$implicit.text, '\n        ');
        if (import3.checkBinding(throwOnChange, this._expr_6, currVal_6)) {
            this.renderer.setText(this._text_3, currVal_6);
            this._expr_6 = currVal_6;
        }
    };
    View_DataTablePagerComponent1.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_DataTablePagerComponent1.prototype.handleEvent_2 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.context.selectPage(this.context.$implicit.number) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_DataTablePagerComponent1;
}(import2.AppView));
var renderType_DataTablePagerComponent = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, styles_DataTablePagerComponent, {});
export var View_DataTablePagerComponent0 = (function (_super) {
    __extends(View_DataTablePagerComponent0, _super);
    function View_DataTablePagerComponent0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DataTablePagerComponent0, renderType_DataTablePagerComponent, import6.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckOnce);
        this._expr_41 = import1.UNINITIALIZED;
        this._expr_42 = import1.UNINITIALIZED;
        this._expr_43 = import1.UNINITIALIZED;
        this._expr_44 = import1.UNINITIALIZED;
        this._expr_45 = import1.UNINITIALIZED;
        this._expr_46 = import1.UNINITIALIZED;
        this._expr_47 = import1.UNINITIALIZED;
        this._expr_48 = import1.UNINITIALIZED;
    }
    View_DataTablePagerComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this._el_1 = import3.createRenderElement(this.renderer, parentRenderNode, 'ul', new import3.InlineArray2(2, 'class', 'pager'), null);
        this._text_2 = this.renderer.createText(this._el_1, '\n      ', null);
        this._el_3 = import3.createRenderElement(this.renderer, this._el_1, 'li', import3.EMPTY_INLINE_ARRAY, null);
        this._text_4 = this.renderer.createText(this._el_3, '\n        ', null);
        this._el_5 = import3.createRenderElement(this.renderer, this._el_3, 'a', new import3.InlineArray2(2, 'href', 'javascript:void(0)'), null);
        this._text_6 = this.renderer.createText(this._el_5, '\n          ', null);
        this._el_7 = import3.createRenderElement(this.renderer, this._el_5, 'i', import3.EMPTY_INLINE_ARRAY, null);
        this._text_8 = this.renderer.createText(this._el_5, '\n        ', null);
        this._text_9 = this.renderer.createText(this._el_3, '\n      ', null);
        this._text_10 = this.renderer.createText(this._el_1, '\n      ', null);
        this._el_11 = import3.createRenderElement(this.renderer, this._el_1, 'li', import3.EMPTY_INLINE_ARRAY, null);
        this._text_12 = this.renderer.createText(this._el_11, '\n        ', null);
        this._el_13 = import3.createRenderElement(this.renderer, this._el_11, 'a', new import3.InlineArray2(2, 'href', 'javascript:void(0)'), null);
        this._text_14 = this.renderer.createText(this._el_13, '\n          ', null);
        this._el_15 = import3.createRenderElement(this.renderer, this._el_13, 'i', import3.EMPTY_INLINE_ARRAY, null);
        this._text_16 = this.renderer.createText(this._el_13, '\n        ', null);
        this._text_17 = this.renderer.createText(this._el_11, '\n      ', null);
        this._text_18 = this.renderer.createText(this._el_1, '\n      ', null);
        this._anchor_19 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._vc_19 = new import9.ViewContainer(19, 1, this, this._anchor_19);
        this._TemplateRef_19_5 = new import11.TemplateRef_(this, 19, this._anchor_19);
        this._NgFor_19_6 = new import10.Wrapper_NgFor(this._vc_19.vcRef, this._TemplateRef_19_5, this.parentView.injectorGet(import12.IterableDiffers, this.parentIndex), this.ref);
        this._text_20 = this.renderer.createText(this._el_1, '\n      ', null);
        this._el_21 = import3.createRenderElement(this.renderer, this._el_1, 'li', import3.EMPTY_INLINE_ARRAY, null);
        this._text_22 = this.renderer.createText(this._el_21, '\n        ', null);
        this._el_23 = import3.createRenderElement(this.renderer, this._el_21, 'a', new import3.InlineArray2(2, 'href', 'javascript:void(0)'), null);
        this._text_24 = this.renderer.createText(this._el_23, '\n          ', null);
        this._el_25 = import3.createRenderElement(this.renderer, this._el_23, 'i', import3.EMPTY_INLINE_ARRAY, null);
        this._text_26 = this.renderer.createText(this._el_23, '\n        ', null);
        this._text_27 = this.renderer.createText(this._el_21, '\n      ', null);
        this._text_28 = this.renderer.createText(this._el_1, '\n      ', null);
        this._el_29 = import3.createRenderElement(this.renderer, this._el_1, 'li', import3.EMPTY_INLINE_ARRAY, null);
        this._text_30 = this.renderer.createText(this._el_29, '\n        ', null);
        this._el_31 = import3.createRenderElement(this.renderer, this._el_29, 'a', new import3.InlineArray2(2, 'href', 'javascript:void(0)'), null);
        this._text_32 = this.renderer.createText(this._el_31, '\n          ', null);
        this._el_33 = import3.createRenderElement(this.renderer, this._el_31, 'i', import3.EMPTY_INLINE_ARRAY, null);
        this._text_34 = this.renderer.createText(this._el_31, '\n        ', null);
        this._text_35 = this.renderer.createText(this._el_29, '\n      ', null);
        this._text_36 = this.renderer.createText(this._el_1, '\n    ', null);
        this._text_37 = this.renderer.createText(parentRenderNode, '\n  ', null);
        var disposable_0 = import3.subscribeToRenderElement(this, this._el_5, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_5));
        var disposable_1 = import3.subscribeToRenderElement(this, this._el_13, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_13));
        var disposable_2 = import3.subscribeToRenderElement(this, this._el_23, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_23));
        var disposable_3 = import3.subscribeToRenderElement(this, this._el_31, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_31));
        this.init(null, (this.renderer.directRenderer ? null : [
            this._text_0,
            this._el_1,
            this._text_2,
            this._el_3,
            this._text_4,
            this._el_5,
            this._text_6,
            this._el_7,
            this._text_8,
            this._text_9,
            this._text_10,
            this._el_11,
            this._text_12,
            this._el_13,
            this._text_14,
            this._el_15,
            this._text_16,
            this._text_17,
            this._text_18,
            this._anchor_19,
            this._text_20,
            this._el_21,
            this._text_22,
            this._el_23,
            this._text_24,
            this._el_25,
            this._text_26,
            this._text_27,
            this._text_28,
            this._el_29,
            this._text_30,
            this._el_31,
            this._text_32,
            this._el_33,
            this._text_34,
            this._text_35,
            this._text_36,
            this._text_37
        ]), [
            disposable_0,
            disposable_1,
            disposable_2,
            disposable_3
        ]);
        return null;
    };
    View_DataTablePagerComponent0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import11.TemplateRef) && (19 === requestNodeIndex))) {
            return this._TemplateRef_19_5;
        }
        if (((token === import13.NgFor) && (19 === requestNodeIndex))) {
            return this._NgFor_19_6.context;
        }
        return notFoundResult;
    };
    View_DataTablePagerComponent0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_19_0_0 = this.context.pages;
        this._NgFor_19_6.check_ngForOf(currVal_19_0_0, throwOnChange, false);
        this._NgFor_19_6.ngDoCheck(this, this._anchor_19, throwOnChange);
        this._vc_19.detectChangesInNestedViews(throwOnChange);
        var currVal_41 = !this.context.canPrevious();
        if (import3.checkBinding(throwOnChange, this._expr_41, currVal_41)) {
            this.renderer.setElementClass(this._el_3, 'disabled', currVal_41);
            this._expr_41 = currVal_41;
        }
        var currVal_42 = import3.inlineInterpolate(1, '', this.context.pagerPreviousIcon, '');
        if (import3.checkBinding(throwOnChange, this._expr_42, currVal_42)) {
            this.renderer.setElementProperty(this._el_7, 'className', currVal_42);
            this._expr_42 = currVal_42;
        }
        var currVal_43 = !this.context.canPrevious();
        if (import3.checkBinding(throwOnChange, this._expr_43, currVal_43)) {
            this.renderer.setElementClass(this._el_11, 'disabled', currVal_43);
            this._expr_43 = currVal_43;
        }
        var currVal_44 = import3.inlineInterpolate(1, '', this.context.pagerLeftArrowIcon, '');
        if (import3.checkBinding(throwOnChange, this._expr_44, currVal_44)) {
            this.renderer.setElementProperty(this._el_15, 'className', currVal_44);
            this._expr_44 = currVal_44;
        }
        var currVal_45 = !this.context.canNext();
        if (import3.checkBinding(throwOnChange, this._expr_45, currVal_45)) {
            this.renderer.setElementClass(this._el_21, 'disabled', currVal_45);
            this._expr_45 = currVal_45;
        }
        var currVal_46 = import3.inlineInterpolate(1, '', this.context.pagerRightArrowIcon, '');
        if (import3.checkBinding(throwOnChange, this._expr_46, currVal_46)) {
            this.renderer.setElementProperty(this._el_25, 'className', currVal_46);
            this._expr_46 = currVal_46;
        }
        var currVal_47 = !this.context.canNext();
        if (import3.checkBinding(throwOnChange, this._expr_47, currVal_47)) {
            this.renderer.setElementClass(this._el_29, 'disabled', currVal_47);
            this._expr_47 = currVal_47;
        }
        var currVal_48 = import3.inlineInterpolate(1, '', this.context.pagerNextIcon, '');
        if (import3.checkBinding(throwOnChange, this._expr_48, currVal_48)) {
            this.renderer.setElementProperty(this._el_33, 'className', currVal_48);
            this._expr_48 = currVal_48;
        }
    };
    View_DataTablePagerComponent0.prototype.destroyInternal = function () {
        this._vc_19.destroyNestedViews();
    };
    View_DataTablePagerComponent0.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 19)) {
            return new View_DataTablePagerComponent1(this.viewUtils, this, 19, this._anchor_19, this._vc_19);
        }
        return null;
    };
    View_DataTablePagerComponent0.prototype.handleEvent_5 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.context.selectPage(1) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    View_DataTablePagerComponent0.prototype.handleEvent_13 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.context.prevPage() !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    View_DataTablePagerComponent0.prototype.handleEvent_23 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.context.nextPage() !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    View_DataTablePagerComponent0.prototype.handleEvent_31 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.context.selectPage(this.context.totalPages) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_DataTablePagerComponent0;
}(import2.AppView));
