// @swimlane/ngx-datatable/release/components/body/body-row-wrapper.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@swimlane/ngx-datatable/release/components/body/body-row-wrapper.component';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/constants';
import * as import8 from '@angular/core/src/linker/component_factory';
import * as import9 from '@angular/core/src/linker/view_container';
import * as import10 from '../../../../../@angular/common/src/directives/ng_template_outlet.ngfactory';
import * as import11 from '@angular/core/src/linker/template_ref';
import * as import12 from '@angular/common/src/directives/ng_template_outlet';
import * as import13 from '../../../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import14 from '@angular/common/src/directives/ng_if';
import * as import15 from '@angular/core/src/security';
export var Wrapper_DataTableRowWrapperComponent = (function () {
    function Wrapper_DataTableRowWrapperComponent() {
        this._changed = false;
        this.context = new import0.DataTableRowWrapperComponent();
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
        this._expr_3 = import1.UNINITIALIZED;
    }
    Wrapper_DataTableRowWrapperComponent.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_DataTableRowWrapperComponent.prototype.ngOnDestroy = function () {
        (this.subscription0 && this.subscription0.unsubscribe());
    };
    Wrapper_DataTableRowWrapperComponent.prototype.check_rowDetailTemplate = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.rowDetailTemplate = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_DataTableRowWrapperComponent.prototype.check_detailRowHeight = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.detailRowHeight = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_DataTableRowWrapperComponent.prototype.check_expanded = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.expanded = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_DataTableRowWrapperComponent.prototype.check_row = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_3, currValue))) {
            this._changed = true;
            this.context.row = currValue;
            this._expr_3 = currValue;
        }
    };
    Wrapper_DataTableRowWrapperComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_DataTableRowWrapperComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_DataTableRowWrapperComponent.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        if ((eventName == 'contextmenu')) {
            var pd_sub_0 = (this.context.onContextmenu($event) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    Wrapper_DataTableRowWrapperComponent.prototype.subscribe = function (view, _eventHandler, emit0) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.rowContextmenu.subscribe(_eventHandler.bind(view, 'rowContextmenu')));
        }
    };
    return Wrapper_DataTableRowWrapperComponent;
}());
var renderType_DataTableRowWrapperComponent_Host = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, [], {});
var View_DataTableRowWrapperComponent_Host0 = (function (_super) {
    __extends(View_DataTableRowWrapperComponent_Host0, _super);
    function View_DataTableRowWrapperComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DataTableRowWrapperComponent_Host0, renderType_DataTableRowWrapperComponent_Host, import6.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways);
    }
    View_DataTableRowWrapperComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'datatable-row-wrapper', new import3.InlineArray2(2, 'class', 'datatable-row-wrapper'), rootSelector, null);
        this.compView_0 = new View_DataTableRowWrapperComponent0(this.viewUtils, this, 0, this._el_0);
        this._DataTableRowWrapperComponent_0_3 = new Wrapper_DataTableRowWrapperComponent();
        this.compView_0.create(this._DataTableRowWrapperComponent_0_3.context);
        var disposable_0 = import3.subscribeToRenderElement(this, this._el_0, new import3.InlineArray2(2, 'contextmenu', null), this.eventHandler(this.handleEvent_0));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), [disposable_0]);
        return new import8.ComponentRef_(0, this, this._el_0, this._DataTableRowWrapperComponent_0_3.context);
    };
    View_DataTableRowWrapperComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.DataTableRowWrapperComponent) && (0 === requestNodeIndex))) {
            return this._DataTableRowWrapperComponent_0_3.context;
        }
        return notFoundResult;
    };
    View_DataTableRowWrapperComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this._DataTableRowWrapperComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange);
        this.compView_0.internalDetectChanges(throwOnChange);
    };
    View_DataTableRowWrapperComponent_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
        this._DataTableRowWrapperComponent_0_3.ngOnDestroy();
    };
    View_DataTableRowWrapperComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_DataTableRowWrapperComponent_Host0.prototype.visitProjectableNodesInternal = function (nodeIndex, ngContentIndex, cb, ctx) {
        if (((nodeIndex == 0) && (ngContentIndex == 0))) { }
    };
    View_DataTableRowWrapperComponent_Host0.prototype.handleEvent_0 = function (eventName, $event) {
        this.compView_0.markPathToRootAsCheckOnce();
        var result = true;
        result = (this._DataTableRowWrapperComponent_0_3.handleEvent(eventName, $event) && result);
        return result;
    };
    return View_DataTableRowWrapperComponent_Host0;
}(import2.AppView));
export var DataTableRowWrapperComponentNgFactory = new import8.ComponentFactory('datatable-row-wrapper', View_DataTableRowWrapperComponent_Host0, import0.DataTableRowWrapperComponent);
var styles_DataTableRowWrapperComponent = [];
var View_DataTableRowWrapperComponent3 = (function (_super) {
    __extends(View_DataTableRowWrapperComponent3, _super);
    function View_DataTableRowWrapperComponent3(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DataTableRowWrapperComponent3, renderType_DataTableRowWrapperComponent, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
    }
    View_DataTableRowWrapperComponent3.prototype.createInternal = function (rootSelector) {
        this._text_0 = this.renderer.createText(null, '\n      ', null);
        this.init(this._text_0, (this.renderer.directRenderer ? null : [this._text_0]), null);
        return null;
    };
    View_DataTableRowWrapperComponent3.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._text_0, ctx);
    };
    return View_DataTableRowWrapperComponent3;
}(import2.AppView));
var View_DataTableRowWrapperComponent2 = (function (_super) {
    __extends(View_DataTableRowWrapperComponent2, _super);
    function View_DataTableRowWrapperComponent2(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DataTableRowWrapperComponent2, renderType_DataTableRowWrapperComponent, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._map_5 = import3.pureProxy1(function (p0) {
            return { row: p0 };
        });
    }
    View_DataTableRowWrapperComponent2.prototype.createInternal = function (rootSelector) {
        this._anchor_0 = this.renderer.createTemplateAnchor(null, null);
        this._vc_0 = new import9.ViewContainer(0, null, this, this._anchor_0);
        this._TemplateRef_0_5 = new import11.TemplateRef_(this, 0, this._anchor_0);
        this._NgTemplateOutlet_0_6 = new import10.Wrapper_NgTemplateOutlet(this._vc_0.vcRef);
        this._el_1 = this.renderer.createTemplateAnchor(null, null);
        this.init(this._el_1, (this.renderer.directRenderer ? null : [this._anchor_0]), null);
        return null;
    };
    View_DataTableRowWrapperComponent2.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import11.TemplateRef) && (0 === requestNodeIndex))) {
            return this._TemplateRef_0_5;
        }
        if (((token === import12.NgTemplateOutlet) && (0 === requestNodeIndex))) {
            return this._NgTemplateOutlet_0_6.context;
        }
        return notFoundResult;
    };
    View_DataTableRowWrapperComponent2.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0_0_0 = this._map_5(this.parentView.parentView.context.row);
        this._NgTemplateOutlet_0_6.check_ngOutletContext(currVal_0_0_0, throwOnChange, false);
        var currVal_0_0_1 = this.parentView.parentView.context.rowDetailTemplate;
        this._NgTemplateOutlet_0_6.check_ngTemplateOutlet(currVal_0_0_1, throwOnChange, false);
        this._NgTemplateOutlet_0_6.ngDoCheck(this, this._anchor_0, throwOnChange);
        this._vc_0.detectChangesInNestedViews(throwOnChange);
    };
    View_DataTableRowWrapperComponent2.prototype.destroyInternal = function () {
        this._vc_0.destroyNestedViews();
    };
    View_DataTableRowWrapperComponent2.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._vc_0.nativeElement, ctx);
        this._vc_0.visitNestedViewRootNodes(cb, ctx);
        cb(this._el_1, ctx);
    };
    View_DataTableRowWrapperComponent2.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 0)) {
            return new View_DataTableRowWrapperComponent3(this.viewUtils, this, 0, this._anchor_0, this._vc_0);
        }
        return null;
    };
    return View_DataTableRowWrapperComponent2;
}(import2.AppView));
var View_DataTableRowWrapperComponent1 = (function (_super) {
    __extends(View_DataTableRowWrapperComponent1, _super);
    function View_DataTableRowWrapperComponent1(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DataTableRowWrapperComponent1, renderType_DataTableRowWrapperComponent, import6.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_7 = import1.UNINITIALIZED;
    }
    View_DataTableRowWrapperComponent1.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.createRenderElement(this.renderer, null, 'div', new import3.InlineArray2(2, 'class', 'datatable-row-detail'), null);
        this._text_1 = this.renderer.createText(this._el_0, '\n      ', null);
        this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._vc_2 = new import9.ViewContainer(2, 0, this, this._anchor_2);
        this._TemplateRef_2_5 = new import11.TemplateRef_(this, 2, this._anchor_2);
        this._NgIf_2_6 = new import13.Wrapper_NgIf(this._vc_2.vcRef, this._TemplateRef_2_5);
        this._text_3 = this.renderer.createText(this._el_0, '\n    ', null);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._anchor_2,
            this._text_3
        ]), null);
        return null;
    };
    View_DataTableRowWrapperComponent1.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import11.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import14.NgIf) && (2 === requestNodeIndex))) {
            return this._NgIf_2_6.context;
        }
        return notFoundResult;
    };
    View_DataTableRowWrapperComponent1.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2_0_0 = this.parentView.context.rowDetailTemplate;
        this._NgIf_2_6.check_ngIf(currVal_2_0_0, throwOnChange, false);
        this._NgIf_2_6.ngDoCheck(this, this._anchor_2, throwOnChange);
        this._vc_2.detectChangesInNestedViews(throwOnChange);
        var currVal_7 = this.parentView.context.detailRowHeight;
        if (import3.checkBinding(throwOnChange, this._expr_7, currVal_7)) {
            this.renderer.setElementStyle(this._el_0, 'height', ((this.viewUtils.sanitizer.sanitize(import15.SecurityContext.STYLE, currVal_7) == null) ? null : (this.viewUtils.sanitizer.sanitize(import15.SecurityContext.STYLE, currVal_7).toString() + 'px')));
            this._expr_7 = currVal_7;
        }
    };
    View_DataTableRowWrapperComponent1.prototype.destroyInternal = function () {
        this._vc_2.destroyNestedViews();
    };
    View_DataTableRowWrapperComponent1.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_DataTableRowWrapperComponent1.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 2)) {
            return new View_DataTableRowWrapperComponent2(this.viewUtils, this, 2, this._anchor_2, this._vc_2);
        }
        return null;
    };
    return View_DataTableRowWrapperComponent1;
}(import2.AppView));
var renderType_DataTableRowWrapperComponent = import3.createRenderComponentType('', 1, import5.ViewEncapsulation.None, styles_DataTableRowWrapperComponent, {});
export var View_DataTableRowWrapperComponent0 = (function (_super) {
    __extends(View_DataTableRowWrapperComponent0, _super);
    function View_DataTableRowWrapperComponent0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DataTableRowWrapperComponent0, renderType_DataTableRowWrapperComponent, import6.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways);
    }
    View_DataTableRowWrapperComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this.projectNodes(parentRenderNode, 0);
        this._text_1 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this._anchor_2 = this.renderer.createTemplateAnchor(parentRenderNode, null);
        this._vc_2 = new import9.ViewContainer(2, null, this, this._anchor_2);
        this._TemplateRef_2_5 = new import11.TemplateRef_(this, 2, this._anchor_2);
        this._NgIf_2_6 = new import13.Wrapper_NgIf(this._vc_2.vcRef, this._TemplateRef_2_5);
        this._text_3 = this.renderer.createText(parentRenderNode, '\n  ', null);
        this.init(null, (this.renderer.directRenderer ? null : [
            this._text_0,
            this._text_1,
            this._anchor_2,
            this._text_3
        ]), null);
        return null;
    };
    View_DataTableRowWrapperComponent0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import11.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import14.NgIf) && (2 === requestNodeIndex))) {
            return this._NgIf_2_6.context;
        }
        return notFoundResult;
    };
    View_DataTableRowWrapperComponent0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2_0_0 = this.context.expanded;
        this._NgIf_2_6.check_ngIf(currVal_2_0_0, throwOnChange, false);
        this._NgIf_2_6.ngDoCheck(this, this._anchor_2, throwOnChange);
        this._vc_2.detectChangesInNestedViews(throwOnChange);
    };
    View_DataTableRowWrapperComponent0.prototype.destroyInternal = function () {
        this._vc_2.destroyNestedViews();
    };
    View_DataTableRowWrapperComponent0.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 2)) {
            return new View_DataTableRowWrapperComponent1(this.viewUtils, this, 2, this._anchor_2, this._vc_2);
        }
        return null;
    };
    return View_DataTableRowWrapperComponent0;
}(import2.AppView));
