// @swimlane/ngx-datatable/release/components/body/body-cell.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@swimlane/ngx-datatable/release/components/body/body-cell.component';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/security';
import * as import6 from '@angular/core/src/metadata/view';
import * as import7 from '@angular/core/src/linker/view_type';
import * as import8 from '@angular/core/src/change_detection/constants';
import * as import9 from '@angular/core/src/linker/component_factory';
import * as import10 from '@angular/core/src/linker/element_ref';
import * as import11 from '@angular/core/src/linker/view_container';
import * as import12 from '../../../../../@angular/common/src/directives/ng_template_outlet.ngfactory';
import * as import13 from '@angular/core/src/linker/template_ref';
import * as import14 from '@angular/common/src/directives/ng_template_outlet';
import * as import15 from '../../../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import16 from '@angular/common/src/directives/ng_if';
export var Wrapper_DataTableBodyCellComponent = (function () {
    function Wrapper_DataTableBodyCellComponent(p0) {
        this._changed = false;
        this.context = new import0.DataTableBodyCellComponent(p0);
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
        this._expr_3 = import1.UNINITIALIZED;
        this._expr_4 = import1.UNINITIALIZED;
        this._expr_5 = import1.UNINITIALIZED;
        this._expr_6 = import1.UNINITIALIZED;
        this._expr_7 = import1.UNINITIALIZED;
        this._expr_8 = import1.UNINITIALIZED;
        this._expr_9 = import1.UNINITIALIZED;
        this._expr_10 = import1.UNINITIALIZED;
        this._expr_11 = import1.UNINITIALIZED;
    }
    Wrapper_DataTableBodyCellComponent.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_DataTableBodyCellComponent.prototype.ngOnDestroy = function () {
        (this.subscription0 && this.subscription0.unsubscribe());
    };
    Wrapper_DataTableBodyCellComponent.prototype.check_row = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.row = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_DataTableBodyCellComponent.prototype.check_column = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.column = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_DataTableBodyCellComponent.prototype.check_rowHeight = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.rowHeight = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_DataTableBodyCellComponent.prototype.check_isSelected = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_3, currValue))) {
            this._changed = true;
            this.context.isSelected = currValue;
            this._expr_3 = currValue;
        }
    };
    Wrapper_DataTableBodyCellComponent.prototype.check_sorts = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_4, currValue))) {
            this._changed = true;
            this.context.sorts = currValue;
            this._expr_4 = currValue;
        }
    };
    Wrapper_DataTableBodyCellComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_DataTableBodyCellComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
        var currVal_5 = this.context.columnCssClasses;
        if (import3.checkBinding(throwOnChange, this._expr_5, currVal_5)) {
            view.renderer.setElementProperty(el, 'className', currVal_5);
            this._expr_5 = currVal_5;
        }
        var currVal_6 = this.context.isFocused;
        if (import3.checkBinding(throwOnChange, this._expr_6, currVal_6)) {
            view.renderer.setElementClass(el, 'active', currVal_6);
            this._expr_6 = currVal_6;
        }
        var currVal_7 = this.context.isSortActive;
        if (import3.checkBinding(throwOnChange, this._expr_7, currVal_7)) {
            view.renderer.setElementClass(el, 'sort-active', currVal_7);
            this._expr_7 = currVal_7;
        }
        var currVal_8 = this.context.isSortAscending;
        if (import3.checkBinding(throwOnChange, this._expr_8, currVal_8)) {
            view.renderer.setElementClass(el, 'sort-asc', currVal_8);
            this._expr_8 = currVal_8;
        }
        var currVal_9 = this.context.isSortDescending;
        if (import3.checkBinding(throwOnChange, this._expr_9, currVal_9)) {
            view.renderer.setElementClass(el, 'sort-desc', currVal_9);
            this._expr_9 = currVal_9;
        }
        var currVal_10 = this.context.width;
        if (import3.checkBinding(throwOnChange, this._expr_10, currVal_10)) {
            view.renderer.setElementStyle(el, 'width', ((view.viewUtils.sanitizer.sanitize(import4.SecurityContext.STYLE, currVal_10) == null) ? null : (view.viewUtils.sanitizer.sanitize(import4.SecurityContext.STYLE, currVal_10).toString() + 'px')));
            this._expr_10 = currVal_10;
        }
        var currVal_11 = this.context.height;
        if (import3.checkBinding(throwOnChange, this._expr_11, currVal_11)) {
            view.renderer.setElementStyle(el, 'height', ((view.viewUtils.sanitizer.sanitize(import4.SecurityContext.STYLE, currVal_11) == null) ? null : view.viewUtils.sanitizer.sanitize(import4.SecurityContext.STYLE, currVal_11).toString()));
            this._expr_11 = currVal_11;
        }
    };
    Wrapper_DataTableBodyCellComponent.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        if ((eventName == 'focus')) {
            var pd_sub_0 = (this.context.onFocus() !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'blur')) {
            var pd_sub_1 = (this.context.onBlur() !== false);
            result = (pd_sub_1 && result);
        }
        if ((eventName == 'click')) {
            var pd_sub_2 = (this.context.onClick($event) !== false);
            result = (pd_sub_2 && result);
        }
        if ((eventName == 'dblclick')) {
            var pd_sub_3 = (this.context.onDblClick($event) !== false);
            result = (pd_sub_3 && result);
        }
        if ((eventName == 'keydown')) {
            var pd_sub_4 = (this.context.onKeyDown($event) !== false);
            result = (pd_sub_4 && result);
        }
        return result;
    };
    Wrapper_DataTableBodyCellComponent.prototype.subscribe = function (view, _eventHandler, emit0) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.activate.subscribe(_eventHandler.bind(view, 'activate')));
        }
    };
    return Wrapper_DataTableBodyCellComponent;
}());
var renderType_DataTableBodyCellComponent_Host = import3.createRenderComponentType('', 0, import6.ViewEncapsulation.None, [], {});
var View_DataTableBodyCellComponent_Host0 = (function (_super) {
    __extends(View_DataTableBodyCellComponent_Host0, _super);
    function View_DataTableBodyCellComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DataTableBodyCellComponent_Host0, renderType_DataTableBodyCellComponent_Host, import7.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import8.ChangeDetectorStatus.CheckAlways);
    }
    View_DataTableBodyCellComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'datatable-body-cell', new import3.InlineArray2(2, 'class', 'datatable-body-cell'), rootSelector, null);
        this.compView_0 = new View_DataTableBodyCellComponent0(this.viewUtils, this, 0, this._el_0);
        this._DataTableBodyCellComponent_0_3 = new Wrapper_DataTableBodyCellComponent(new import10.ElementRef(this._el_0));
        this.compView_0.create(this._DataTableBodyCellComponent_0_3.context);
        var disposable_0 = import3.subscribeToRenderElement(this, this._el_0, new import3.InlineArray16(10, 'focus', null, 'blur', null, 'click', null, 'dblclick', null, 'keydown', null), this.eventHandler(this.handleEvent_0));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), [disposable_0]);
        return new import9.ComponentRef_(0, this, this._el_0, this._DataTableBodyCellComponent_0_3.context);
    };
    View_DataTableBodyCellComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.DataTableBodyCellComponent) && (0 === requestNodeIndex))) {
            return this._DataTableBodyCellComponent_0_3.context;
        }
        return notFoundResult;
    };
    View_DataTableBodyCellComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this._DataTableBodyCellComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange);
        this._DataTableBodyCellComponent_0_3.checkHost(this, this.compView_0, this._el_0, throwOnChange);
        this.compView_0.internalDetectChanges(throwOnChange);
    };
    View_DataTableBodyCellComponent_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
        this._DataTableBodyCellComponent_0_3.ngOnDestroy();
    };
    View_DataTableBodyCellComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_DataTableBodyCellComponent_Host0.prototype.handleEvent_0 = function (eventName, $event) {
        this.compView_0.markPathToRootAsCheckOnce();
        var result = true;
        result = (this._DataTableBodyCellComponent_0_3.handleEvent(eventName, $event) && result);
        return result;
    };
    return View_DataTableBodyCellComponent_Host0;
}(import2.AppView));
export var DataTableBodyCellComponentNgFactory = new import9.ComponentFactory('datatable-body-cell', View_DataTableBodyCellComponent_Host0, import0.DataTableBodyCellComponent);
var styles_DataTableBodyCellComponent = [];
var View_DataTableBodyCellComponent1 = (function (_super) {
    __extends(View_DataTableBodyCellComponent1, _super);
    function View_DataTableBodyCellComponent1(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DataTableBodyCellComponent1, renderType_DataTableBodyCellComponent, import7.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import8.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_4 = import1.UNINITIALIZED;
    }
    View_DataTableBodyCellComponent1.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.createRenderElement(this.renderer, null, 'label', new import3.InlineArray2(2, 'class', 'datatable-checkbox'), null);
        this._text_1 = this.renderer.createText(this._el_0, '\n        ', null);
        this._el_2 = import3.createRenderElement(this.renderer, this._el_0, 'input', new import3.InlineArray2(2, 'type', 'checkbox'), null);
        this._text_3 = this.renderer.createText(this._el_0, '\n      ', null);
        var disposable_0 = import3.subscribeToRenderElement(this, this._el_2, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_2));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._el_2,
            this._text_3
        ]), [disposable_0]);
        return null;
    };
    View_DataTableBodyCellComponent1.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_4 = this.parentView.context.isSelected;
        if (import3.checkBinding(throwOnChange, this._expr_4, currVal_4)) {
            this.renderer.setElementProperty(this._el_2, 'checked', currVal_4);
            this._expr_4 = currVal_4;
        }
    };
    View_DataTableBodyCellComponent1.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_DataTableBodyCellComponent1.prototype.handleEvent_2 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.context.onCheckboxChange($event) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_DataTableBodyCellComponent1;
}(import2.AppView));
var View_DataTableBodyCellComponent2 = (function (_super) {
    __extends(View_DataTableBodyCellComponent2, _super);
    function View_DataTableBodyCellComponent2(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DataTableBodyCellComponent2, renderType_DataTableBodyCellComponent, import7.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import8.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_2 = import1.UNINITIALIZED;
    }
    View_DataTableBodyCellComponent2.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.createRenderElement(this.renderer, null, 'span', import3.EMPTY_INLINE_ARRAY, null);
        this._text_1 = this.renderer.createText(this._el_0, '\n      ', null);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1
        ]), null);
        return null;
    };
    View_DataTableBodyCellComponent2.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2 = this.parentView.context.value;
        if (import3.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setElementProperty(this._el_0, 'innerHTML', this.viewUtils.sanitizer.sanitize(import4.SecurityContext.HTML, currVal_2));
            this._expr_2 = currVal_2;
        }
    };
    View_DataTableBodyCellComponent2.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_DataTableBodyCellComponent2;
}(import2.AppView));
var View_DataTableBodyCellComponent4 = (function (_super) {
    __extends(View_DataTableBodyCellComponent4, _super);
    function View_DataTableBodyCellComponent4(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DataTableBodyCellComponent4, renderType_DataTableBodyCellComponent, import7.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import8.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
    }
    View_DataTableBodyCellComponent4.prototype.createInternal = function (rootSelector) {
        this._text_0 = this.renderer.createText(null, '\n      ', null);
        this.init(this._text_0, (this.renderer.directRenderer ? null : [this._text_0]), null);
        return null;
    };
    View_DataTableBodyCellComponent4.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._text_0, ctx);
    };
    return View_DataTableBodyCellComponent4;
}(import2.AppView));
var View_DataTableBodyCellComponent3 = (function (_super) {
    __extends(View_DataTableBodyCellComponent3, _super);
    function View_DataTableBodyCellComponent3(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DataTableBodyCellComponent3, renderType_DataTableBodyCellComponent, import7.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import8.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._map_5 = import3.pureProxy3(function (p0, p1, p2) {
            return {
                value: p0,
                row: p1,
                column: p2
            };
        });
    }
    View_DataTableBodyCellComponent3.prototype.createInternal = function (rootSelector) {
        this._anchor_0 = this.renderer.createTemplateAnchor(null, null);
        this._vc_0 = new import11.ViewContainer(0, null, this, this._anchor_0);
        this._TemplateRef_0_5 = new import13.TemplateRef_(this, 0, this._anchor_0);
        this._NgTemplateOutlet_0_6 = new import12.Wrapper_NgTemplateOutlet(this._vc_0.vcRef);
        this._el_1 = this.renderer.createTemplateAnchor(null, null);
        this.init(this._el_1, (this.renderer.directRenderer ? null : [this._anchor_0]), null);
        return null;
    };
    View_DataTableBodyCellComponent3.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import13.TemplateRef) && (0 === requestNodeIndex))) {
            return this._TemplateRef_0_5;
        }
        if (((token === import14.NgTemplateOutlet) && (0 === requestNodeIndex))) {
            return this._NgTemplateOutlet_0_6.context;
        }
        return notFoundResult;
    };
    View_DataTableBodyCellComponent3.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0_0_0 = this._map_5(this.parentView.context.value, this.parentView.context.row, this.parentView.context.column);
        this._NgTemplateOutlet_0_6.check_ngOutletContext(currVal_0_0_0, throwOnChange, false);
        var currVal_0_0_1 = this.parentView.context.column.cellTemplate;
        this._NgTemplateOutlet_0_6.check_ngTemplateOutlet(currVal_0_0_1, throwOnChange, false);
        this._NgTemplateOutlet_0_6.ngDoCheck(this, this._anchor_0, throwOnChange);
        this._vc_0.detectChangesInNestedViews(throwOnChange);
    };
    View_DataTableBodyCellComponent3.prototype.destroyInternal = function () {
        this._vc_0.destroyNestedViews();
    };
    View_DataTableBodyCellComponent3.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._vc_0.nativeElement, ctx);
        this._vc_0.visitNestedViewRootNodes(cb, ctx);
        cb(this._el_1, ctx);
    };
    View_DataTableBodyCellComponent3.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 0)) {
            return new View_DataTableBodyCellComponent4(this.viewUtils, this, 0, this._anchor_0, this._vc_0);
        }
        return null;
    };
    return View_DataTableBodyCellComponent3;
}(import2.AppView));
var renderType_DataTableBodyCellComponent = import3.createRenderComponentType('', 0, import6.ViewEncapsulation.None, styles_DataTableBodyCellComponent, {});
export var View_DataTableBodyCellComponent0 = (function (_super) {
    __extends(View_DataTableBodyCellComponent0, _super);
    function View_DataTableBodyCellComponent0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DataTableBodyCellComponent0, renderType_DataTableBodyCellComponent, import7.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import8.ChangeDetectorStatus.CheckAlways);
    }
    View_DataTableBodyCellComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this._el_1 = import3.createRenderElement(this.renderer, parentRenderNode, 'div', new import3.InlineArray2(2, 'class', 'datatable-body-cell-label'), null);
        this._text_2 = this.renderer.createText(this._el_1, '\n      ', null);
        this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._vc_3 = new import11.ViewContainer(3, 1, this, this._anchor_3);
        this._TemplateRef_3_5 = new import13.TemplateRef_(this, 3, this._anchor_3);
        this._NgIf_3_6 = new import15.Wrapper_NgIf(this._vc_3.vcRef, this._TemplateRef_3_5);
        this._text_4 = this.renderer.createText(this._el_1, '\n      ', null);
        this._anchor_5 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._vc_5 = new import11.ViewContainer(5, 1, this, this._anchor_5);
        this._TemplateRef_5_5 = new import13.TemplateRef_(this, 5, this._anchor_5);
        this._NgIf_5_6 = new import15.Wrapper_NgIf(this._vc_5.vcRef, this._TemplateRef_5_5);
        this._text_6 = this.renderer.createText(this._el_1, '\n      ', null);
        this._anchor_7 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._vc_7 = new import11.ViewContainer(7, 1, this, this._anchor_7);
        this._TemplateRef_7_5 = new import13.TemplateRef_(this, 7, this._anchor_7);
        this._NgIf_7_6 = new import15.Wrapper_NgIf(this._vc_7.vcRef, this._TemplateRef_7_5);
        this._text_8 = this.renderer.createText(this._el_1, '\n    ', null);
        this._text_9 = this.renderer.createText(parentRenderNode, '\n  ', null);
        this.init(null, (this.renderer.directRenderer ? null : [
            this._text_0,
            this._el_1,
            this._text_2,
            this._anchor_3,
            this._text_4,
            this._anchor_5,
            this._text_6,
            this._anchor_7,
            this._text_8,
            this._text_9
        ]), null);
        return null;
    };
    View_DataTableBodyCellComponent0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import13.TemplateRef) && (3 === requestNodeIndex))) {
            return this._TemplateRef_3_5;
        }
        if (((token === import16.NgIf) && (3 === requestNodeIndex))) {
            return this._NgIf_3_6.context;
        }
        if (((token === import13.TemplateRef) && (5 === requestNodeIndex))) {
            return this._TemplateRef_5_5;
        }
        if (((token === import16.NgIf) && (5 === requestNodeIndex))) {
            return this._NgIf_5_6.context;
        }
        if (((token === import13.TemplateRef) && (7 === requestNodeIndex))) {
            return this._TemplateRef_7_5;
        }
        if (((token === import16.NgIf) && (7 === requestNodeIndex))) {
            return this._NgIf_7_6.context;
        }
        return notFoundResult;
    };
    View_DataTableBodyCellComponent0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_3_0_0 = this.context.column.checkboxable;
        this._NgIf_3_6.check_ngIf(currVal_3_0_0, throwOnChange, false);
        this._NgIf_3_6.ngDoCheck(this, this._anchor_3, throwOnChange);
        var currVal_5_0_0 = !this.context.column.cellTemplate;
        this._NgIf_5_6.check_ngIf(currVal_5_0_0, throwOnChange, false);
        this._NgIf_5_6.ngDoCheck(this, this._anchor_5, throwOnChange);
        var currVal_7_0_0 = this.context.column.cellTemplate;
        this._NgIf_7_6.check_ngIf(currVal_7_0_0, throwOnChange, false);
        this._NgIf_7_6.ngDoCheck(this, this._anchor_7, throwOnChange);
        this._vc_3.detectChangesInNestedViews(throwOnChange);
        this._vc_5.detectChangesInNestedViews(throwOnChange);
        this._vc_7.detectChangesInNestedViews(throwOnChange);
    };
    View_DataTableBodyCellComponent0.prototype.destroyInternal = function () {
        this._vc_3.destroyNestedViews();
        this._vc_5.destroyNestedViews();
        this._vc_7.destroyNestedViews();
    };
    View_DataTableBodyCellComponent0.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 3)) {
            return new View_DataTableBodyCellComponent1(this.viewUtils, this, 3, this._anchor_3, this._vc_3);
        }
        if ((nodeIndex == 5)) {
            return new View_DataTableBodyCellComponent2(this.viewUtils, this, 5, this._anchor_5, this._vc_5);
        }
        if ((nodeIndex == 7)) {
            return new View_DataTableBodyCellComponent3(this.viewUtils, this, 7, this._anchor_7, this._vc_7);
        }
        return null;
    };
    return View_DataTableBodyCellComponent0;
}(import2.AppView));
