// @swimlane/ngx-datatable/release/components/body/progress-bar.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@swimlane/ngx-datatable/release/components/body/progress-bar.component';
import * as import1 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/metadata/view';
import * as import5 from '@angular/core/src/linker/view_type';
import * as import6 from '@angular/core/src/change_detection/constants';
import * as import7 from '@angular/core/src/linker/component_factory';
export var Wrapper_ProgressBarComponent = (function () {
    function Wrapper_ProgressBarComponent() {
        this._changed = false;
        this.context = new import0.ProgressBarComponent();
    }
    Wrapper_ProgressBarComponent.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_ProgressBarComponent.prototype.ngOnDestroy = function () {
    };
    Wrapper_ProgressBarComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_ProgressBarComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_ProgressBarComponent.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_ProgressBarComponent.prototype.subscribe = function (view, _eventHandler) {
        this._eventHandler = _eventHandler;
    };
    return Wrapper_ProgressBarComponent;
}());
var renderType_ProgressBarComponent_Host = import3.createRenderComponentType('', 0, import4.ViewEncapsulation.None, [], {});
var View_ProgressBarComponent_Host0 = (function (_super) {
    __extends(View_ProgressBarComponent_Host0, _super);
    function View_ProgressBarComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_ProgressBarComponent_Host0, renderType_ProgressBarComponent_Host, import5.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import6.ChangeDetectorStatus.CheckAlways);
    }
    View_ProgressBarComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'datatable-progress', import3.EMPTY_INLINE_ARRAY, rootSelector, null);
        this.compView_0 = new View_ProgressBarComponent0(this.viewUtils, this, 0, this._el_0);
        this._ProgressBarComponent_0_3 = new Wrapper_ProgressBarComponent();
        this.compView_0.create(this._ProgressBarComponent_0_3.context);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
        return new import7.ComponentRef_(0, this, this._el_0, this._ProgressBarComponent_0_3.context);
    };
    View_ProgressBarComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.ProgressBarComponent) && (0 === requestNodeIndex))) {
            return this._ProgressBarComponent_0_3.context;
        }
        return notFoundResult;
    };
    View_ProgressBarComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (this._ProgressBarComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange)) {
            this.compView_0.markAsCheckOnce();
        }
        this.compView_0.internalDetectChanges(throwOnChange);
    };
    View_ProgressBarComponent_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
    };
    View_ProgressBarComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_ProgressBarComponent_Host0;
}(import1.AppView));
export var ProgressBarComponentNgFactory = new import7.ComponentFactory('datatable-progress', View_ProgressBarComponent_Host0, import0.ProgressBarComponent);
var styles_ProgressBarComponent = [];
var renderType_ProgressBarComponent = import3.createRenderComponentType('', 0, import4.ViewEncapsulation.None, styles_ProgressBarComponent, {});
export var View_ProgressBarComponent0 = (function (_super) {
    __extends(View_ProgressBarComponent0, _super);
    function View_ProgressBarComponent0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_ProgressBarComponent0, renderType_ProgressBarComponent, import5.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import6.ChangeDetectorStatus.CheckOnce);
    }
    View_ProgressBarComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this._el_1 = import3.createRenderElement(this.renderer, parentRenderNode, 'div', new import3.InlineArray4(4, 'class', 'progress-linear', 'role', 'progressbar'), null);
        this._text_2 = this.renderer.createText(this._el_1, '\n      ', null);
        this._el_3 = import3.createRenderElement(this.renderer, this._el_1, 'div', new import3.InlineArray2(2, 'class', 'container'), null);
        this._text_4 = this.renderer.createText(this._el_3, '\n        ', null);
        this._el_5 = import3.createRenderElement(this.renderer, this._el_3, 'div', new import3.InlineArray2(2, 'class', 'bar'), null);
        this._text_6 = this.renderer.createText(this._el_3, '\n      ', null);
        this._text_7 = this.renderer.createText(this._el_1, '\n    ', null);
        this._text_8 = this.renderer.createText(parentRenderNode, '\n  ', null);
        this.init(null, (this.renderer.directRenderer ? null : [
            this._text_0,
            this._el_1,
            this._text_2,
            this._el_3,
            this._text_4,
            this._el_5,
            this._text_6,
            this._text_7,
            this._text_8
        ]), null);
        return null;
    };
    return View_ProgressBarComponent0;
}(import1.AppView));
