// @swimlane/ngx-datatable/release/components/body/selection.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@swimlane/ngx-datatable/release/components/body/selection.component';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/constants';
import * as import8 from '@angular/core/src/linker/component_factory';
export var Wrapper_DataTableSelectionComponent = (function () {
    function Wrapper_DataTableSelectionComponent() {
        this._changed = false;
        this.context = new import0.DataTableSelectionComponent();
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
        this._expr_3 = import1.UNINITIALIZED;
        this._expr_4 = import1.UNINITIALIZED;
        this._expr_5 = import1.UNINITIALIZED;
    }
    Wrapper_DataTableSelectionComponent.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_DataTableSelectionComponent.prototype.ngOnDestroy = function () {
        (this.subscription0 && this.subscription0.unsubscribe());
        (this.subscription1 && this.subscription1.unsubscribe());
    };
    Wrapper_DataTableSelectionComponent.prototype.check_rows = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.rows = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_DataTableSelectionComponent.prototype.check_selected = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.selected = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_DataTableSelectionComponent.prototype.check_selectEnabled = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.selectEnabled = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_DataTableSelectionComponent.prototype.check_selectionType = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_3, currValue))) {
            this._changed = true;
            this.context.selectionType = currValue;
            this._expr_3 = currValue;
        }
    };
    Wrapper_DataTableSelectionComponent.prototype.check_rowIdentity = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_4, currValue))) {
            this._changed = true;
            this.context.rowIdentity = currValue;
            this._expr_4 = currValue;
        }
    };
    Wrapper_DataTableSelectionComponent.prototype.check_selectCheck = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_5, currValue))) {
            this._changed = true;
            this.context.selectCheck = currValue;
            this._expr_5 = currValue;
        }
    };
    Wrapper_DataTableSelectionComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_DataTableSelectionComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_DataTableSelectionComponent.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_DataTableSelectionComponent.prototype.subscribe = function (view, _eventHandler, emit0, emit1) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.activate.subscribe(_eventHandler.bind(view, 'activate')));
        }
        if (emit1) {
            (this.subscription1 = this.context.select.subscribe(_eventHandler.bind(view, 'select')));
        }
    };
    return Wrapper_DataTableSelectionComponent;
}());
var renderType_DataTableSelectionComponent_Host = import3.createRenderComponentType('', 0, import5.ViewEncapsulation.None, [], {});
var View_DataTableSelectionComponent_Host0 = (function (_super) {
    __extends(View_DataTableSelectionComponent_Host0, _super);
    function View_DataTableSelectionComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DataTableSelectionComponent_Host0, renderType_DataTableSelectionComponent_Host, import6.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways);
    }
    View_DataTableSelectionComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'datatable-selection', import3.EMPTY_INLINE_ARRAY, rootSelector, null);
        this.compView_0 = new View_DataTableSelectionComponent0(this.viewUtils, this, 0, this._el_0);
        this._DataTableSelectionComponent_0_3 = new Wrapper_DataTableSelectionComponent();
        this.compView_0.create(this._DataTableSelectionComponent_0_3.context);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
        return new import8.ComponentRef_(0, this, this._el_0, this._DataTableSelectionComponent_0_3.context);
    };
    View_DataTableSelectionComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.DataTableSelectionComponent) && (0 === requestNodeIndex))) {
            return this._DataTableSelectionComponent_0_3.context;
        }
        return notFoundResult;
    };
    View_DataTableSelectionComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this._DataTableSelectionComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange);
        this.compView_0.internalDetectChanges(throwOnChange);
    };
    View_DataTableSelectionComponent_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
        this._DataTableSelectionComponent_0_3.ngOnDestroy();
    };
    View_DataTableSelectionComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_DataTableSelectionComponent_Host0.prototype.visitProjectableNodesInternal = function (nodeIndex, ngContentIndex, cb, ctx) {
        if (((nodeIndex == 0) && (ngContentIndex == 0))) { }
    };
    return View_DataTableSelectionComponent_Host0;
}(import2.AppView));
export var DataTableSelectionComponentNgFactory = new import8.ComponentFactory('datatable-selection', View_DataTableSelectionComponent_Host0, import0.DataTableSelectionComponent);
var styles_DataTableSelectionComponent = [];
var renderType_DataTableSelectionComponent = import3.createRenderComponentType('', 1, import5.ViewEncapsulation.None, styles_DataTableSelectionComponent, {});
export var View_DataTableSelectionComponent0 = (function (_super) {
    __extends(View_DataTableSelectionComponent0, _super);
    function View_DataTableSelectionComponent0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DataTableSelectionComponent0, renderType_DataTableSelectionComponent, import6.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import7.ChangeDetectorStatus.CheckAlways);
    }
    View_DataTableSelectionComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this.projectNodes(parentRenderNode, 0);
        this._text_1 = this.renderer.createText(parentRenderNode, '\n  ', null);
        this.init(null, (this.renderer.directRenderer ? null : [
            this._text_0,
            this._text_1
        ]), null);
        return null;
    };
    return View_DataTableSelectionComponent0;
}(import2.AppView));
