// @swimlane/ngx-datatable/release/directives/draggable.directive.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from '@swimlane/ngx-datatable/release/directives/draggable.directive';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import3 from '@angular/core/src/linker/view_utils';
export var Wrapper_DraggableDirective = (function () {
    function Wrapper_DraggableDirective(p0) {
        this._changed = false;
        this.context = new import0.DraggableDirective(p0);
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
    }
    Wrapper_DraggableDirective.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_DraggableDirective.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
        (this.subscription0 && this.subscription0.unsubscribe());
        (this.subscription1 && this.subscription1.unsubscribe());
        (this.subscription2 && this.subscription2.unsubscribe());
    };
    Wrapper_DraggableDirective.prototype.check_dragModel = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.dragModel = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_DraggableDirective.prototype.check_dragX = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.dragX = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_DraggableDirective.prototype.check_dragY = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.dragY = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_DraggableDirective.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_DraggableDirective.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_DraggableDirective.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        if ((eventName == 'document:mouseup')) {
            var pd_sub_0 = (this.context.onMouseup($event) !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'mousedown')) {
            var pd_sub_1 = (this.context.onMousedown($event) !== false);
            result = (pd_sub_1 && result);
        }
        return result;
    };
    Wrapper_DraggableDirective.prototype.subscribe = function (view, _eventHandler, emit0, emit1, emit2) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.dragStart.subscribe(_eventHandler.bind(view, 'dragStart')));
        }
        if (emit1) {
            (this.subscription1 = this.context.dragging.subscribe(_eventHandler.bind(view, 'dragging')));
        }
        if (emit2) {
            (this.subscription2 = this.context.dragEnd.subscribe(_eventHandler.bind(view, 'dragEnd')));
        }
    };
    return Wrapper_DraggableDirective;
}());
