// @swimlane/ngx-datatable/release/directives/orderable.directive.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from '@swimlane/ngx-datatable/release/directives/orderable.directive';
export var Wrapper_OrderableDirective = (function () {
    function Wrapper_OrderableDirective(p0) {
        this._changed = false;
        this.context = new import0.OrderableDirective(p0);
    }
    Wrapper_OrderableDirective.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_OrderableDirective.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
        (this.subscription0 && this.subscription0.unsubscribe());
    };
    Wrapper_OrderableDirective.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_OrderableDirective.prototype.checkHost = function (view, componentView, el, throwOnChange) {
    };
    Wrapper_OrderableDirective.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_OrderableDirective.prototype.subscribe = function (view, _eventHandler, emit0) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.reorder.subscribe(_eventHandler.bind(view, 'reorder')));
        }
    };
    return Wrapper_OrderableDirective;
}());
