// @swimlane/ngx-datatable/release/directives/resizeable.directive.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from '@swimlane/ngx-datatable/release/directives/resizeable.directive';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import3 from '@angular/core/src/linker/view_utils';
export var Wrapper_ResizeableDirective = (function () {
    function Wrapper_ResizeableDirective(p0) {
        this._changed = false;
        this.context = new import0.ResizeableDirective(p0);
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
        this._expr_3 = import1.UNINITIALIZED;
    }
    Wrapper_ResizeableDirective.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_ResizeableDirective.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
        (this.subscription0 && this.subscription0.unsubscribe());
    };
    Wrapper_ResizeableDirective.prototype.check_resizeEnabled = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.resizeEnabled = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_ResizeableDirective.prototype.check_minWidth = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.minWidth = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_ResizeableDirective.prototype.check_maxWidth = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.maxWidth = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_ResizeableDirective.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_ResizeableDirective.prototype.checkHost = function (view, componentView, el, throwOnChange) {
        var currVal_3 = this.context.resizeEnabled;
        if (import3.checkBinding(throwOnChange, this._expr_3, currVal_3)) {
            view.renderer.setElementClass(el, 'resizeable', currVal_3);
            this._expr_3 = currVal_3;
        }
    };
    Wrapper_ResizeableDirective.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        if ((eventName == 'document:mouseup')) {
            var pd_sub_0 = (this.context.onMouseup() !== false);
            result = (pd_sub_0 && result);
        }
        if ((eventName == 'mousedown')) {
            var pd_sub_1 = (this.context.onMousedown($event) !== false);
            result = (pd_sub_1 && result);
        }
        return result;
    };
    Wrapper_ResizeableDirective.prototype.subscribe = function (view, _eventHandler, emit0) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.resize.subscribe(_eventHandler.bind(view, 'resize')));
        }
    };
    return Wrapper_ResizeableDirective;
}());
