// @swimlane/ngx-datatable/release/directives/visibility.directive.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from '@swimlane/ngx-datatable/release/directives/visibility.directive';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import3 from '@angular/core/src/linker/view_utils';
export var Wrapper_VisibilityDirective = (function () {
    function Wrapper_VisibilityDirective(p0, p1) {
        this._changed = false;
        this.context = new import0.VisibilityDirective(p0, p1);
        this._expr_0 = import1.UNINITIALIZED;
    }
    Wrapper_VisibilityDirective.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_VisibilityDirective.prototype.ngOnDestroy = function () {
        this.context.ngOnDestroy();
        (this.subscription0 && this.subscription0.unsubscribe());
    };
    Wrapper_VisibilityDirective.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        if (!throwOnChange) {
            if ((view.numberOfChecks === 0)) {
                this.context.ngOnInit();
            }
        }
        return changed;
    };
    Wrapper_VisibilityDirective.prototype.checkHost = function (view, componentView, el, throwOnChange) {
        var currVal_0 = this.context.isVisible;
        if (import3.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            view.renderer.setElementClass(el, 'visible', currVal_0);
            this._expr_0 = currVal_0;
        }
    };
    Wrapper_VisibilityDirective.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_VisibilityDirective.prototype.subscribe = function (view, _eventHandler, emit0) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.visible.subscribe(_eventHandler.bind(view, 'visible')));
        }
    };
    return Wrapper_VisibilityDirective;
}());
