// @swimlane/ngx-datatable/release/components/header/header-cell.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from '@swimlane/ngx-datatable/release/components/header/header-cell.component';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/security';
import * as import6 from '@angular/core/src/metadata/view';
import * as import7 from '@angular/core/src/linker/view_type';
import * as import8 from '@angular/core/src/change_detection/constants';
import * as import9 from '@angular/core/src/linker/component_factory';
import * as import10 from '@angular/core/src/linker/view_container';
import * as import11 from '../../../../../@angular/common/src/directives/ng_template_outlet.ngfactory';
import * as import12 from '@angular/core/src/linker/template_ref';
import * as import13 from '@angular/common/src/directives/ng_template_outlet';
import * as import14 from '../../../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import15 from '@angular/common/src/directives/ng_if';
export var Wrapper_DataTableHeaderCellComponent = (function () {
    function Wrapper_DataTableHeaderCellComponent() {
        this._changed = false;
        this.context = new import0.DataTableHeaderCellComponent();
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this._expr_2 = import1.UNINITIALIZED;
        this._expr_3 = import1.UNINITIALIZED;
        this._expr_4 = import1.UNINITIALIZED;
        this._expr_5 = import1.UNINITIALIZED;
        this._expr_6 = import1.UNINITIALIZED;
        this._expr_7 = import1.UNINITIALIZED;
        this._expr_8 = import1.UNINITIALIZED;
        this._expr_9 = import1.UNINITIALIZED;
        this._expr_10 = import1.UNINITIALIZED;
        this._expr_11 = import1.UNINITIALIZED;
        this._expr_12 = import1.UNINITIALIZED;
        this._expr_13 = import1.UNINITIALIZED;
    }
    Wrapper_DataTableHeaderCellComponent.prototype.ngOnDetach = function (view, componentView, el) {
    };
    Wrapper_DataTableHeaderCellComponent.prototype.ngOnDestroy = function () {
        (this.subscription0 && this.subscription0.unsubscribe());
        (this.subscription1 && this.subscription1.unsubscribe());
    };
    Wrapper_DataTableHeaderCellComponent.prototype.check_sortType = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_0, currValue))) {
            this._changed = true;
            this.context.sortType = currValue;
            this._expr_0 = currValue;
        }
    };
    Wrapper_DataTableHeaderCellComponent.prototype.check_column = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_1, currValue))) {
            this._changed = true;
            this.context.column = currValue;
            this._expr_1 = currValue;
        }
    };
    Wrapper_DataTableHeaderCellComponent.prototype.check_sortAscendingIcon = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_2, currValue))) {
            this._changed = true;
            this.context.sortAscendingIcon = currValue;
            this._expr_2 = currValue;
        }
    };
    Wrapper_DataTableHeaderCellComponent.prototype.check_sortDescendingIcon = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_3, currValue))) {
            this._changed = true;
            this.context.sortDescendingIcon = currValue;
            this._expr_3 = currValue;
        }
    };
    Wrapper_DataTableHeaderCellComponent.prototype.check_allRowsSelected = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_4, currValue))) {
            this._changed = true;
            this.context.allRowsSelected = currValue;
            this._expr_4 = currValue;
        }
    };
    Wrapper_DataTableHeaderCellComponent.prototype.check_selectionType = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_5, currValue))) {
            this._changed = true;
            this.context.selectionType = currValue;
            this._expr_5 = currValue;
        }
    };
    Wrapper_DataTableHeaderCellComponent.prototype.check_headerHeight = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_6, currValue))) {
            this._changed = true;
            this.context.headerHeight = currValue;
            this._expr_6 = currValue;
        }
    };
    Wrapper_DataTableHeaderCellComponent.prototype.check_sorts = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import3.checkBinding(throwOnChange, this._expr_7, currValue))) {
            this._changed = true;
            this.context.sorts = currValue;
            this._expr_7 = currValue;
        }
    };
    Wrapper_DataTableHeaderCellComponent.prototype.ngDoCheck = function (view, el, throwOnChange) {
        var changed = this._changed;
        this._changed = false;
        return changed;
    };
    Wrapper_DataTableHeaderCellComponent.prototype.checkHost = function (view, componentView, el, throwOnChange) {
        var currVal_8 = this.context.headerHeight;
        if (import3.checkBinding(throwOnChange, this._expr_8, currVal_8)) {
            view.renderer.setElementStyle(el, 'height', ((view.viewUtils.sanitizer.sanitize(import4.SecurityContext.STYLE, currVal_8) == null) ? null : (view.viewUtils.sanitizer.sanitize(import4.SecurityContext.STYLE, currVal_8).toString() + 'px')));
            this._expr_8 = currVal_8;
        }
        var currVal_9 = this.context.columnCssClasses;
        if (import3.checkBinding(throwOnChange, this._expr_9, currVal_9)) {
            view.renderer.setElementProperty(el, 'className', currVal_9);
            this._expr_9 = currVal_9;
        }
        var currVal_10 = this.context.name;
        if (import3.checkBinding(throwOnChange, this._expr_10, currVal_10)) {
            view.renderer.setElementAttribute(el, 'title', ((currVal_10 == null) ? null : currVal_10.toString()));
            this._expr_10 = currVal_10;
        }
        var currVal_11 = this.context.minWidth;
        if (import3.checkBinding(throwOnChange, this._expr_11, currVal_11)) {
            view.renderer.setElementStyle(el, 'minWidth', ((view.viewUtils.sanitizer.sanitize(import4.SecurityContext.STYLE, currVal_11) == null) ? null : (view.viewUtils.sanitizer.sanitize(import4.SecurityContext.STYLE, currVal_11).toString() + 'px')));
            this._expr_11 = currVal_11;
        }
        var currVal_12 = this.context.maxWidth;
        if (import3.checkBinding(throwOnChange, this._expr_12, currVal_12)) {
            view.renderer.setElementStyle(el, 'maxWidth', ((view.viewUtils.sanitizer.sanitize(import4.SecurityContext.STYLE, currVal_12) == null) ? null : (view.viewUtils.sanitizer.sanitize(import4.SecurityContext.STYLE, currVal_12).toString() + 'px')));
            this._expr_12 = currVal_12;
        }
        var currVal_13 = this.context.width;
        if (import3.checkBinding(throwOnChange, this._expr_13, currVal_13)) {
            view.renderer.setElementStyle(el, 'width', ((view.viewUtils.sanitizer.sanitize(import4.SecurityContext.STYLE, currVal_13) == null) ? null : (view.viewUtils.sanitizer.sanitize(import4.SecurityContext.STYLE, currVal_13).toString() + 'px')));
            this._expr_13 = currVal_13;
        }
    };
    Wrapper_DataTableHeaderCellComponent.prototype.handleEvent = function (eventName, $event) {
        var result = true;
        return result;
    };
    Wrapper_DataTableHeaderCellComponent.prototype.subscribe = function (view, _eventHandler, emit0, emit1) {
        this._eventHandler = _eventHandler;
        if (emit0) {
            (this.subscription0 = this.context.sort.subscribe(_eventHandler.bind(view, 'sort')));
        }
        if (emit1) {
            (this.subscription1 = this.context.select.subscribe(_eventHandler.bind(view, 'select')));
        }
    };
    return Wrapper_DataTableHeaderCellComponent;
}());
var renderType_DataTableHeaderCellComponent_Host = import3.createRenderComponentType('', 0, import6.ViewEncapsulation.None, [], {});
var View_DataTableHeaderCellComponent_Host0 = (function (_super) {
    __extends(View_DataTableHeaderCellComponent_Host0, _super);
    function View_DataTableHeaderCellComponent_Host0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DataTableHeaderCellComponent_Host0, renderType_DataTableHeaderCellComponent_Host, import7.ViewType.HOST, viewUtils, parentView, parentIndex, parentElement, import8.ChangeDetectorStatus.CheckAlways);
    }
    View_DataTableHeaderCellComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer, 'datatable-header-cell', import3.EMPTY_INLINE_ARRAY, rootSelector, null);
        this.compView_0 = new View_DataTableHeaderCellComponent0(this.viewUtils, this, 0, this._el_0);
        this._DataTableHeaderCellComponent_0_3 = new Wrapper_DataTableHeaderCellComponent();
        this.compView_0.create(this._DataTableHeaderCellComponent_0_3.context);
        this.init(this._el_0, (this.renderer.directRenderer ? null : [this._el_0]), null);
        return new import9.ComponentRef_(0, this, this._el_0, this._DataTableHeaderCellComponent_0_3.context);
    };
    View_DataTableHeaderCellComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.DataTableHeaderCellComponent) && (0 === requestNodeIndex))) {
            return this._DataTableHeaderCellComponent_0_3.context;
        }
        return notFoundResult;
    };
    View_DataTableHeaderCellComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this._DataTableHeaderCellComponent_0_3.ngDoCheck(this, this._el_0, throwOnChange);
        this._DataTableHeaderCellComponent_0_3.checkHost(this, this.compView_0, this._el_0, throwOnChange);
        this.compView_0.internalDetectChanges(throwOnChange);
    };
    View_DataTableHeaderCellComponent_Host0.prototype.destroyInternal = function () {
        this.compView_0.destroy();
        this._DataTableHeaderCellComponent_0_3.ngOnDestroy();
    };
    View_DataTableHeaderCellComponent_Host0.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    return View_DataTableHeaderCellComponent_Host0;
}(import2.AppView));
export var DataTableHeaderCellComponentNgFactory = new import9.ComponentFactory('datatable-header-cell', View_DataTableHeaderCellComponent_Host0, import0.DataTableHeaderCellComponent);
var styles_DataTableHeaderCellComponent = [];
var View_DataTableHeaderCellComponent1 = (function (_super) {
    __extends(View_DataTableHeaderCellComponent1, _super);
    function View_DataTableHeaderCellComponent1(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DataTableHeaderCellComponent1, renderType_DataTableHeaderCellComponent, import7.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import8.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_4 = import1.UNINITIALIZED;
    }
    View_DataTableHeaderCellComponent1.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.createRenderElement(this.renderer, null, 'label', new import3.InlineArray2(2, 'class', 'datatable-checkbox'), null);
        this._text_1 = this.renderer.createText(this._el_0, '\n        ', null);
        this._el_2 = import3.createRenderElement(this.renderer, this._el_0, 'input', new import3.InlineArray2(2, 'type', 'checkbox'), null);
        this._text_3 = this.renderer.createText(this._el_0, '\n      ', null);
        var disposable_0 = import3.subscribeToRenderElement(this, this._el_2, new import3.InlineArray2(2, 'change', null), this.eventHandler(this.handleEvent_2));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1,
            this._el_2,
            this._text_3
        ]), [disposable_0]);
        return null;
    };
    View_DataTableHeaderCellComponent1.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_4 = this.parentView.context.allRowsSelected;
        if (import3.checkBinding(throwOnChange, this._expr_4, currVal_4)) {
            this.renderer.setElementAttribute(this._el_2, 'checked', ((currVal_4 == null) ? null : currVal_4.toString()));
            this._expr_4 = currVal_4;
        }
    };
    View_DataTableHeaderCellComponent1.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_DataTableHeaderCellComponent1.prototype.handleEvent_2 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'change')) {
            var pd_sub_0 = (this.parentView.context.select.emit(!this.parentView.context.allRowsSelected) !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_DataTableHeaderCellComponent1;
}(import2.AppView));
var View_DataTableHeaderCellComponent2 = (function (_super) {
    __extends(View_DataTableHeaderCellComponent2, _super);
    function View_DataTableHeaderCellComponent2(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DataTableHeaderCellComponent2, renderType_DataTableHeaderCellComponent, import7.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import8.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._expr_2 = import1.UNINITIALIZED;
    }
    View_DataTableHeaderCellComponent2.prototype.createInternal = function (rootSelector) {
        this._el_0 = import3.createRenderElement(this.renderer, null, 'span', new import3.InlineArray2(2, 'class', 'datatable-header-cell-label draggable'), null);
        this._text_1 = this.renderer.createText(this._el_0, '\n      ', null);
        var disposable_0 = import3.subscribeToRenderElement(this, this._el_0, new import3.InlineArray2(2, 'click', null), this.eventHandler(this.handleEvent_0));
        this.init(this._el_0, (this.renderer.directRenderer ? null : [
            this._el_0,
            this._text_1
        ]), [disposable_0]);
        return null;
    };
    View_DataTableHeaderCellComponent2.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2 = this.parentView.context.name;
        if (import3.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setElementProperty(this._el_0, 'innerHTML', this.viewUtils.sanitizer.sanitize(import4.SecurityContext.HTML, currVal_2));
            this._expr_2 = currVal_2;
        }
    };
    View_DataTableHeaderCellComponent2.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._el_0, ctx);
    };
    View_DataTableHeaderCellComponent2.prototype.handleEvent_0 = function (eventName, $event) {
        this.markPathToRootAsCheckOnce();
        var result = true;
        if ((eventName == 'click')) {
            var pd_sub_0 = (this.parentView.context.onSort() !== false);
            result = (pd_sub_0 && result);
        }
        return result;
    };
    return View_DataTableHeaderCellComponent2;
}(import2.AppView));
var View_DataTableHeaderCellComponent4 = (function (_super) {
    __extends(View_DataTableHeaderCellComponent4, _super);
    function View_DataTableHeaderCellComponent4(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DataTableHeaderCellComponent4, renderType_DataTableHeaderCellComponent, import7.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import8.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
    }
    View_DataTableHeaderCellComponent4.prototype.createInternal = function (rootSelector) {
        this._text_0 = this.renderer.createText(null, '\n      ', null);
        this.init(this._text_0, (this.renderer.directRenderer ? null : [this._text_0]), null);
        return null;
    };
    View_DataTableHeaderCellComponent4.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._text_0, ctx);
    };
    return View_DataTableHeaderCellComponent4;
}(import2.AppView));
var View_DataTableHeaderCellComponent3 = (function (_super) {
    __extends(View_DataTableHeaderCellComponent3, _super);
    function View_DataTableHeaderCellComponent3(viewUtils, parentView, parentIndex, parentElement, declaredViewContainer) {
        _super.call(this, View_DataTableHeaderCellComponent3, renderType_DataTableHeaderCellComponent, import7.ViewType.EMBEDDED, viewUtils, parentView, parentIndex, parentElement, import8.ChangeDetectorStatus.CheckAlways, declaredViewContainer);
        this._map_5 = import3.pureProxy2(function (p0, p1) {
            return {
                column: p0,
                sortDir: p1
            };
        });
    }
    View_DataTableHeaderCellComponent3.prototype.createInternal = function (rootSelector) {
        this._anchor_0 = this.renderer.createTemplateAnchor(null, null);
        this._vc_0 = new import10.ViewContainer(0, null, this, this._anchor_0);
        this._TemplateRef_0_5 = new import12.TemplateRef_(this, 0, this._anchor_0);
        this._NgTemplateOutlet_0_6 = new import11.Wrapper_NgTemplateOutlet(this._vc_0.vcRef);
        this._el_1 = this.renderer.createTemplateAnchor(null, null);
        this.init(this._el_1, (this.renderer.directRenderer ? null : [this._anchor_0]), null);
        return null;
    };
    View_DataTableHeaderCellComponent3.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import12.TemplateRef) && (0 === requestNodeIndex))) {
            return this._TemplateRef_0_5;
        }
        if (((token === import13.NgTemplateOutlet) && (0 === requestNodeIndex))) {
            return this._NgTemplateOutlet_0_6.context;
        }
        return notFoundResult;
    };
    View_DataTableHeaderCellComponent3.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0_0_0 = this._map_5(this.parentView.context.column, this.parentView.context.sortDir);
        this._NgTemplateOutlet_0_6.check_ngOutletContext(currVal_0_0_0, throwOnChange, false);
        var currVal_0_0_1 = this.parentView.context.column.headerTemplate;
        this._NgTemplateOutlet_0_6.check_ngTemplateOutlet(currVal_0_0_1, throwOnChange, false);
        this._NgTemplateOutlet_0_6.ngDoCheck(this, this._anchor_0, throwOnChange);
        this._vc_0.detectChangesInNestedViews(throwOnChange);
    };
    View_DataTableHeaderCellComponent3.prototype.destroyInternal = function () {
        this._vc_0.destroyNestedViews();
    };
    View_DataTableHeaderCellComponent3.prototype.visitRootNodesInternal = function (cb, ctx) {
        cb(this._vc_0.nativeElement, ctx);
        this._vc_0.visitNestedViewRootNodes(cb, ctx);
        cb(this._el_1, ctx);
    };
    View_DataTableHeaderCellComponent3.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 0)) {
            return new View_DataTableHeaderCellComponent4(this.viewUtils, this, 0, this._anchor_0, this._vc_0);
        }
        return null;
    };
    return View_DataTableHeaderCellComponent3;
}(import2.AppView));
var renderType_DataTableHeaderCellComponent = import3.createRenderComponentType('', 0, import6.ViewEncapsulation.None, styles_DataTableHeaderCellComponent, {});
export var View_DataTableHeaderCellComponent0 = (function (_super) {
    __extends(View_DataTableHeaderCellComponent0, _super);
    function View_DataTableHeaderCellComponent0(viewUtils, parentView, parentIndex, parentElement) {
        _super.call(this, View_DataTableHeaderCellComponent0, renderType_DataTableHeaderCellComponent, import7.ViewType.COMPONENT, viewUtils, parentView, parentIndex, parentElement, import8.ChangeDetectorStatus.CheckAlways);
        this._expr_22 = import1.UNINITIALIZED;
    }
    View_DataTableHeaderCellComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.parentElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this._el_1 = import3.createRenderElement(this.renderer, parentRenderNode, 'div', import3.EMPTY_INLINE_ARRAY, null);
        this._text_2 = this.renderer.createText(this._el_1, '\n      ', null);
        this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._vc_3 = new import10.ViewContainer(3, 1, this, this._anchor_3);
        this._TemplateRef_3_5 = new import12.TemplateRef_(this, 3, this._anchor_3);
        this._NgIf_3_6 = new import14.Wrapper_NgIf(this._vc_3.vcRef, this._TemplateRef_3_5);
        this._text_4 = this.renderer.createText(this._el_1, '\n      ', null);
        this._anchor_5 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._vc_5 = new import10.ViewContainer(5, 1, this, this._anchor_5);
        this._TemplateRef_5_5 = new import12.TemplateRef_(this, 5, this._anchor_5);
        this._NgIf_5_6 = new import14.Wrapper_NgIf(this._vc_5.vcRef, this._TemplateRef_5_5);
        this._text_6 = this.renderer.createText(this._el_1, '\n      ', null);
        this._anchor_7 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._vc_7 = new import10.ViewContainer(7, 1, this, this._anchor_7);
        this._TemplateRef_7_5 = new import12.TemplateRef_(this, 7, this._anchor_7);
        this._NgIf_7_6 = new import14.Wrapper_NgIf(this._vc_7.vcRef, this._TemplateRef_7_5);
        this._text_8 = this.renderer.createText(this._el_1, '\n      ', null);
        this._el_9 = import3.createRenderElement(this.renderer, this._el_1, 'span', new import3.InlineArray2(2, 'class', 'sort-btn'), null);
        this._text_10 = this.renderer.createText(this._el_9, '\n      ', null);
        this._text_11 = this.renderer.createText(this._el_1, '\n    ', null);
        this._text_12 = this.renderer.createText(parentRenderNode, '\n  ', null);
        this.init(null, (this.renderer.directRenderer ? null : [
            this._text_0,
            this._el_1,
            this._text_2,
            this._anchor_3,
            this._text_4,
            this._anchor_5,
            this._text_6,
            this._anchor_7,
            this._text_8,
            this._el_9,
            this._text_10,
            this._text_11,
            this._text_12
        ]), null);
        return null;
    };
    View_DataTableHeaderCellComponent0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import12.TemplateRef) && (3 === requestNodeIndex))) {
            return this._TemplateRef_3_5;
        }
        if (((token === import15.NgIf) && (3 === requestNodeIndex))) {
            return this._NgIf_3_6.context;
        }
        if (((token === import12.TemplateRef) && (5 === requestNodeIndex))) {
            return this._TemplateRef_5_5;
        }
        if (((token === import15.NgIf) && (5 === requestNodeIndex))) {
            return this._NgIf_5_6.context;
        }
        if (((token === import12.TemplateRef) && (7 === requestNodeIndex))) {
            return this._TemplateRef_7_5;
        }
        if (((token === import15.NgIf) && (7 === requestNodeIndex))) {
            return this._NgIf_7_6.context;
        }
        return notFoundResult;
    };
    View_DataTableHeaderCellComponent0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_3_0_0 = this.context.isCheckboxable;
        this._NgIf_3_6.check_ngIf(currVal_3_0_0, throwOnChange, false);
        this._NgIf_3_6.ngDoCheck(this, this._anchor_3, throwOnChange);
        var currVal_5_0_0 = !this.context.column.headerTemplate;
        this._NgIf_5_6.check_ngIf(currVal_5_0_0, throwOnChange, false);
        this._NgIf_5_6.ngDoCheck(this, this._anchor_5, throwOnChange);
        var currVal_7_0_0 = this.context.column.headerTemplate;
        this._NgIf_7_6.check_ngIf(currVal_7_0_0, throwOnChange, false);
        this._NgIf_7_6.ngDoCheck(this, this._anchor_7, throwOnChange);
        this._vc_3.detectChangesInNestedViews(throwOnChange);
        this._vc_5.detectChangesInNestedViews(throwOnChange);
        this._vc_7.detectChangesInNestedViews(throwOnChange);
        var currVal_22 = this.context.sortClass;
        if (import3.checkBinding(throwOnChange, this._expr_22, currVal_22)) {
            this.renderer.setElementProperty(this._el_9, 'className', currVal_22);
            this._expr_22 = currVal_22;
        }
    };
    View_DataTableHeaderCellComponent0.prototype.destroyInternal = function () {
        this._vc_3.destroyNestedViews();
        this._vc_5.destroyNestedViews();
        this._vc_7.destroyNestedViews();
    };
    View_DataTableHeaderCellComponent0.prototype.createEmbeddedViewInternal = function (nodeIndex) {
        if ((nodeIndex == 3)) {
            return new View_DataTableHeaderCellComponent1(this.viewUtils, this, 3, this._anchor_3, this._vc_3);
        }
        if ((nodeIndex == 5)) {
            return new View_DataTableHeaderCellComponent2(this.viewUtils, this, 5, this._anchor_5, this._vc_5);
        }
        if ((nodeIndex == 7)) {
            return new View_DataTableHeaderCellComponent3(this.viewUtils, this, 7, this._anchor_7, this._vc_7);
        }
        return null;
    };
    return View_DataTableHeaderCellComponent0;
}(import2.AppView));
